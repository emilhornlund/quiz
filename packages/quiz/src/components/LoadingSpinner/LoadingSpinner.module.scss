@import '../../styles/colors';
@import '../../styles/helpers';
@import '../../styles/loading';

.loadingSpinnerContainer {
  @extend .loading-spinner;

  @include mobile {
    $--mobile-loading-spinner-size: pxToRem(80px);
    @include loading-spinner-dimensions($--mobile-loading-spinner-size, $white, default-mobile);
  }

  @include tablet {
    $--tablet-loading-spinner-size: pxToRem(100px);
    @include loading-spinner-dimensions($--tablet-loading-spinner-size, $white, default-tablet);
  }

  @include desktop {
    $--desktop-loading-spinner-size: pxToRem(120px);
    @include loading-spinner-dimensions($--desktop-loading-spinner-size, $white, default-desktop);
  }
}
