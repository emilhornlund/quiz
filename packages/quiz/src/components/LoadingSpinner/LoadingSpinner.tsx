import React, { FC } from 'react'

import styles from './LoadingSpinner.module.scss'

const LoadingSpinner: FC = () => (
  <div className={styles.loadingSpinnerContainer}>
    <div></div>
    <div></div>
    <div></div>
  </div>
)

export default LoadingSpinner
