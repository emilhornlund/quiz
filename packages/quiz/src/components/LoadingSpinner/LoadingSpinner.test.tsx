import { render } from '@testing-library/react'
import React from 'react'
import { describe, expect, it } from 'vitest'

import LoadingSpinner from './LoadingSpinner'

describe('LoadingSpinner', () => {
  it('should render a LoadingSpinner with default props', async () => {
    const { container } = render(<LoadingSpinner />)

    expect(container).toMatchSnapshot()
  })
})
