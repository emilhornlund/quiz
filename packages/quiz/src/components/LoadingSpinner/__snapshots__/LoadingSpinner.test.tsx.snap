// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSpinner > should render a LoadingSpinner with default props 1`] = `
<div>
  <div
    class="loadingSpinnerContainer"
  >
    <div />
    <div />
    <div />
  </div>
</div>
`;
